File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/uncomment/direct/Java/Python/s824230885_commented.py
import sys

class InputIterator:
    def __init__(self):
        self.input_line = [line.strip() for line in sys.stdin.readlines()]
        self.index = 0
        self.max = len(self.input_line)

    def hasNext(self):
        return self.index < self.max

    def next(self):
        if self.hasNext():
            return self.input_line[self.index++]
        else:
            raise IndexError("There is no more input")

ii = InputIterator()
out = sys.stdout.write

def flush():
    sys.stdout.flush()

def myout(t):
    out(f"{t}\n")
    flush()

def myerr(t):
    sys.stderr.write(f"debug:{t}\n")

def next():
    return ii.next()

def hasNext():
    return ii.hasNext()

def nextInt():
    return int(next())

def nextLong():
    return int(next())

def nextDouble():
    return float(next())

def nextStrArray():
    return next().split(" ")

def nextCharArray():
    return list(next())

def nextIntArray():
    return [int(x) for x in nextStrArray()]

def nextLongArray():
    return [int(x) for x in nextStrArray()]

def myconv(lst, no):
    if no == 8:
        return " ".join(lst)
    elif no == 9:
        return "\n".join(lst)
    elif no == 0:
        return "".join(lst)
    else:
        raise TypeError("Don't join")

def solve():
    N = nextInt()
    lst = nextIntArray()
    max_val = 0
    output = 0
    for i in range(N):
        if max_val <= lst[i]:
            max_val = lst[i]
            output += 1
    myout(output)

solve()
flush()  # End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/uncomment/direct/Java/C++/s824230885_commented.cpp
#include <iostream>
#include <vector>
#include <sstream>

using namespace std;

class InputIterator {
public:
    vector<string> inputLine;
    int index = 0;
    string read;

    InputIterator() {
        string line;
        while(getline(cin, line)) {
            inputLine.push_back(line);
        }
    }

    bool hasNext() { return (index < inputLine.size()); }
    string next() {
        if(hasNext()) {
            return inputLine[index++];
        } else {
            throw out_of_range("There is no more input");
        }
    }
};

InputIterator ii;
ostream& out = cout;

void flush() { out.flush(); }
void myout(const auto& t) { out << t << endl; }
void myerr(const auto& t) { cerr << "debug:" << t << endl; }
string next() { return ii.next(); }
bool hasNext() { return ii.hasNext(); }
int nextInt() { return stoi(next()); }
long long nextLong() { return stoll(next()); }
double nextDouble() { return stod(next()); }
vector<string> nextStrArray() { return split(next(), ' '); }
vector<string> nextCharArray() { return split(next(), '\0'); }
vector<int> nextIntArray() {
    vector<string> input = nextStrArray();
    vector<int> ret;
    for(const auto& s : input) {
        ret.push_back(stoi(s));
    }
    return ret;
}
vector<long long> nextLongArray() {
    vector<string> input = nextStrArray();
    vector<long long> ret;
    for(const auto& s : input) {
        ret.push_back(stoll(s));
    }
    return ret;
}

vector<string> split(const string& str, char delimiter) {
    stringstream ss(str);
    string token;
    vector<string> tokens;
    while(getline(ss, token, delimiter)) {
        tokens.push_back(token);
    }
    return tokens;
}

void solve() {
    int N = nextInt();
    vector<int> list = nextIntArray();
    int maxVal = 0;
    int output = 0;
    for(int i = 0; i < N; i++) {
        if(maxVal <= list[i]) {
            maxVal = list[i];
            output++;
        }
    }
    myout(output);
}

int main() {
    solve();
    flush();
} // End of Code
